Eplant/views/eFP/index.tsx

```diff
@@ -28,120 +28,24 @@ import {
-  return (
-    <Popper transition anchorEl={props.el} open={open}>
-      {({ TransitionProps }) => (
-        <Grow {...TransitionProps} timeout={350}>
-          <Box
-            sx={(theme) => ({
-              backgroundColor: theme.palette.background.transparentOverlay,
-              backdropFilter: 'blur(7px)',
-              boxShadow: theme.shadows[3],
-              borderRadius: 1,
-            })}
-          >
-            <Table size="small">
-              <TableBody>
-                <TableRow>
-                  <TableCell
-                    sx={{
-                      color: theme.palette.secondary.main,
-                      textAlign: 'right',
-                    }}
-                  >
-                    Sample name
-                  </TableCell>
-                  <TableCell>{props.tissue.name}</TableCell>
-                </TableRow>
-                <TableRow>
-                  <TableCell
-                    sx={{
-                      color: theme.palette.secondary.main,
-                      textAlign: 'right',
-                    }}
-                  >
-                    Level
-                  </TableCell>
-                  <TableCell>
-                    {props.tissue.mean.toFixed(2)}±{props.tissue.std.toFixed(2)}
-                  </TableCell>
-                </TableRow>
-                <TableRow>
-                  <TableCell
-                    sx={{
-                      color: theme.palette.secondary.main,
-                      textAlign: 'right',
-                    }}
-                  >
-                    Samples
-                  </TableCell>
-                  <TableCell>{props.tissue.samples}</TableCell>
-                </TableRow>
-                <TableRow>
-                  <TableCell
-                    sx={{
-                      color: theme.palette.secondary.main,
-                      textAlign: 'right',
-                      borderBottom: 'none',
-                    }}
-                  >
-                    Log2 fold change vs control
-                  </TableCell>
-                  <TableCell sx={{ borderBottom: 'none' }}>
-                    {Math.log2(
-                      props.tissue.mean / (props.data.control ?? 1)
-                    ).toFixed(2)}
-                  </TableCell>
-                </TableRow>
-              </TableBody>
-            </Table>
-          </Box>
-        </Grow>
-      )}
-    </Popper>
-  )
-}
 
 export default class EFP implements View<EFPData, EFPState, EFPAction> {
   getInitialState: () => EFPState = () => ({
     colorMode: 'absolute',
     renderAsThumbnail: false,
   })
+  tooltipComponent: (props: { el: SVGElement | null; group: EFPGroup; tissue: EFPTissue; data: EFPData }) => React.JSX.Element
   constructor(
     public name: string,
     public id: EFPId,
     public svgURL: string,
     public xmlURL: string
   ) {
     this.component = this.component.bind(this)
+    this.tooltipComponent = SVGTooltip;
```

Comment:
Is there a reason to not use the raw `SVGToolip` component? I think aligning the property names over multiple components sounds like more work than controlling the type of tooltip through a property on one component

Reply:
Since different EFP views may need different tooltips (I'm not sure how likely it is to need to implement another EFP view but the Cell EFP is an example of the tooltips being different) we needed a way to either generalize the tooltip component or be able to assign different tooltip components to different views.

I looked into generalizing the tooltip component and having the different views pass in some object representation to it as props, however this idea gets a bit complicated since the values being displayed by the tooltip is only known at runtime. It is still possible to do this, but it seems a bit more complicated than necessary. It seems a bit easier to just define a new tooltip component whenever needed (even though it does lead to some copy-pasted code which isn't ideal).